sdk/lib/crt/startup/crtexe.c

```diff
@@ -324,6 +326,7 @@ __tmainCRTStartup (void)
 #endif
     mainret = main (argc, argv, envp);
 #endif
+    __call_atexit();
```

Comment:
isnt the dll responsible for calling this?
or is the crtexe for the case where the crt is linked into the exe?

Reply:
atexit is executed on a per-module base. First the registered entries within the exe are processed (from the code tat is linked into the exe), then msvcrt (and any other loaded dll) processes it's own entries (te corresponding __call_atexit is in crtdll.c). This is why exitfunc2 is last in the sequence, since it was registered through msvcrt.dll, not in the current module.